tests/fsstress/fsstress_test.go

```diff
@@ -0,0 +1,346 @@
+
+	tc.AddProcInfo(edgeNode, checkAppRunning(appName))
+
+	appID, err := uuid.FromString(appInstanceConfig.Uuidandversion.Uuid)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	callback := func() {
+		fmt.Printf("--- app %s logs ---\n", appInstanceConfig.Displayname)
+		if err = tc.GetController().LogAppsChecker(edgeNode.GetID(), appID, nil, eapps.HandleFactory(eapps.LogJSON, false), eapps.LogExist, 0); err != nil {
+			t.Fatalf("LogAppsChecker: %s", err)
+		}
+		fmt.Println("------")
+	}
+
+	tc.WaitForProcWithErrorCallback(int(timewait.Seconds()), callback)
+}
+
+func getAppInstanceConfig(edgeNode *device.Ctx, appName string) *config.AppInstanceConfig {
+
+	var appInstanceConfig *config.AppInstanceConfig
+
+	for _, id := range edgeNode.GetApplicationInstances() {
+		appConfig, _ := tc.GetController().GetApplicationInstanceConfig(id)
+		if appConfig.Displayname == appName {
+			appInstanceConfig = appConfig
+			break
+		}
+	}
+	return appInstanceConfig
+}
+
+
+//TestAccess checks if SSH of app is accessible
+func TestAccess(t *testing.T) {
+
+	edgeNode := tc.GetEdgeNode(tc.WithTest(t))
+
+	setAppName()
+
+	appInstanceConfig := getAppInstanceConfig(edgeNode, appName)
+
+	if appInstanceConfig == nil {
+		t.Fatalf("No app found with name %s", appName)
+	}
+
+	appID, err := uuid.FromString(appInstanceConfig.Uuidandversion.Uuid)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	t.Log(utils.AddTimestamp("Add function to obtain EVE IP"))
+
+	tc.AddProcTimer(edgeNode, getEVEIP(edgeNode))
+
+	t.Log(utils.AddTimestamp("Add trying to access SSH of app"))
+
+	tc.AddProcTimer(edgeNode, projects.SendCommandSSH(&externalIP, sshPort, "ubuntu", "passw0rd", "exit", true, defaults.DefaultRepeatTimeout))
+
+	tc.ExpandOnSuccess(int(expand.Seconds()))
+
+	callback := func() {
+		fmt.Printf("--- app %s logs ---\n", appInstanceConfig.Displayname)
+		if err = tc.GetController().LogAppsChecker(edgeNode.GetID(), appID, nil, eapps.HandleFactory(eapps.LogJSON, false), eapps.LogExist, 0); err != nil {
+			t.Fatalf("LogAppsChecker: %s", err)
+		}
+		fmt.Println("------")
+	}
+
+	tc.WaitForProcWithErrorCallback(int(timewait.Seconds()), callback)
+
+}
+
+//TestRunStress run fsstress test on guest vm
+func TestRunStress(t *testing.T) {
+	edgeNode := tc.GetEdgeNode(tc.WithTest(t))
+	setAppName()
+
+	appInstanceConfig := getAppInstanceConfig(edgeNode, appName)
+
+	if appInstanceConfig == nil {
+		t.Fatalf("No app found with name %s", appName)
+	}
+
+	appID, err := uuid.FromString(appInstanceConfig.Uuidandversion.Uuid)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ex, err := os.Executable()
+	if err != nil {
+			panic(fmt.Sprintf("AAAA %v",err))
+	}
+
+	dir := filepath.Dir(ex)
+	patch := filepath.Join(dir, "/../../", "tests/fsstress/testdata/run-script.sh")
+
+	tc.AddProcTimer(edgeNode, getEVEIP(edgeNode))
+
+	t.Log(utils.AddTimestamp("Send script on guest VM"))
+	tc.AddProcTimer(edgeNode, projects.SendCommandSCP(&externalIP, sshPort, "ubuntu", "passw0rd", patch , "/home/ubuntu/run-script.sh"))
```

Comment:
I am probably just missing something here, but how do we know that SCP finished successfully?

Reply:
Actually from Eden perspective it will retry to call function we provided as the second argument to AddProcTimer until it will return not-nill error (which will indicates success of processing). It sounds strange, but it is.
But let me check what it will do in case of no timeout I see few lines below.

Reply:
Now there, if something goes wrong while transferring the file, the test will panic and end